src/mavsdk/plugins/mission/mission_impl.cpp

```diff
@@ -554,6 +587,70 @@ MissionImpl::convert_to_int_items(const std::vector<MissionItem>& mission_items)
             int_items.push_back(next_item);
         }
 
+        if (item.vehicle_action != VehicleAction::None &&
+            item.vehicle_action != VehicleAction::Takeoff) {
+            // There is a vehicle action that we need to send.
+
+            // Current is the 0th waypoint
+            uint8_t current = ((int_items.size() == 0) ? 1 : 0);
+
+            uint8_t autocontinue = 1;
+
+            uint16_t command = 0;
+            float param1 = NAN;
+            float param2 = NAN;
+            float param3 = NAN;
+            switch (item.vehicle_action) {
+                case VehicleAction::Land:
+                    command = MAV_CMD_NAV_LAND; // Land at current position with same heading
+                    break;
+                case VehicleAction::TransitionToFw:
+                    command = MAV_CMD_DO_VTOL_TRANSITION; // Do transition
+                    param1 = MAV_VTOL_STATE_FW; // Target state is Fixed-Wing
+                    param2 = 0; // Normal transition
+                    break;
+                case VehicleAction::TransitionToMc:
+                    command = MAV_CMD_DO_VTOL_TRANSITION;
+                    param1 = MAV_VTOL_STATE_MC; // Target state is Multi-Copter
+                    param2 = 0; // Normal transition
+                    break;
+                default:
+                    LogErr() << "Error: vehicle action not supported";
+                    break;
+            }
+
+            const int32_t x = int32_t(std::round(item.latitude_deg * 1e7));
+            const int32_t y = int32_t(std::round(item.longitude_deg * 1e7));
+            float z = item.relative_altitude_m;
+            MAV_FRAME frame = MAV_FRAME_GLOBAL_RELATIVE_ALT_INT;
+
+            if (command == MAV_CMD_NAV_LAND) {
+                z = 0;
```

Comment:
I wonder what happens if the landing spot is below the takeoff/home altitude. I think PX4 just ignores this and lands anyway.

Reply:
Actually, this altitude should be an indication of the landing altitude, so presumably we should use the altitude field here.

Reply:
So you suggest to assign min value of float to z variable, so that vehicle will descent until it will reach ground?

Reply:
Hm, so the altitude set here is the altitude of the waypoint before landing, I see. In that case we really don't know the landing altitude and 0 is probably the best guess. I'd say we leave it as is then.

Reply:
Ok